1608130: Document how to update android-components (#667)
mozilla/glean: docs/dev/cut-a-new-release.md

    * This is important so that no changes are lost.
    * This might have merge conflicts with the `master` branch, which you need to fix before it is merged.
8. Once the above pull request lands, delete the support branch.

## Upgrading android-components to a new version of Glean

> On Android, many users of Glean import it through its wrapper in [`android-components`](https://github.com/mozilla-mobile/android-components).

Reviewer: ```suggestion
On Android, Mozilla products consume the Glean SDK through its wrapper in [`android-components`](https://github.com/mozilla-mobile/android-components).
```

Reviewer: This is actually a requirement for Mozilla products :-)

> Therefore, when a new Glean release is made, `android-components` must also be updated.

Reviewer: ```suggestion
Therefore, when a new Glean SDK release is made, `android-components` must also be updated.
```

> After following one of the above instructions to make a Glean release:

Reviewer: ```suggestion
After following one of the above instructions to make a Glean SDK release:
```

> 1) Ensure that CI has completed and the artifacts have published to [the Glean GitHub releases page](https://github.com/mozilla/glean/releases/).

Reviewer 2: While this seems to render as a list just fine, should we use the name `<number>.` format as for other lists in this document, for consistency?

2) Publish Glean to [Mozilla's Maven repository](https://maven.mozilla.org/). 
   This is currently a manual step that can only performed by individuals with the necessary credentials.
   Ask in `#releaseduty-mobile` on Mozilla's internal Slack for assistance.
   
3) Create a pull request against `android-components` to update the Glean version with the following changes:
   

>    - The Glean version is updated in the `mozilla_glean` variable in the `buildSrc/src/main/java/Dependencies.kt` file.

Reviewer: ```suggestion
   - The Glean version is updated in the `mozilla_glean` variable in the [`buildSrc/src/main/java/Dependencies.kt`](https://github.com/mozilla-mobile/android-components/blob/69546999739ab19d21425e9a98e107e438a3f905/buildSrc/src/main/java/Dependencies.kt#L32) file.
```

>    - The relevant parts of the Glean changelog copied into the `android-components` changelog.

Reviewer: ```suggestion
   - The relevant parts of the Glean changelog copied into the [`android-components` changelog](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md), in the top section with the`-SNAPSHOT` suffix in the title.
```

Author: That's only relevant if they are releasing snapshots, which IIUC is not the long term plan.  I'd be fine with just saying "top section" here, though.

Reviewer: Sounds good to me!
